test/meson.build

```diff
@@ -0,0 +1,41 @@
+test_src_files = [
+	'dbus.c',
+	'draw.c',
+	'dunst.c',
+	'helpers.c',
+	'icon-lookup.c',
+	'icon.c',
+	'ini.c',
+	'input.c',
+	'log.c',
+	'markup.c',
+	'menu.c',
+	'misc.c',
+	'notification.c',
+	'option_parser.c',
+	'queues.c',
+	'rules.c',
+	'setting.c',
+	'settings_data.c',
+	'test.c',
+	'utils.c',
+]
+
+fs = import('fs')
+foreach dunst_src_file : dunst_src_files
+	if fs.name(dunst_src_file) not in test_src_files
+		test_src_files += dunst_src_file
+	endif
+endforeach
+
+test_prog = executable('test-runner',
+	test_src_files,
+	dependencies: dunst_depends,
+	c_args: c_version_arg,
+	install: false,
```

Comment:
`build_by_default: false` to tell meson that this target is only needed when running `meson test` or `ninja test`, and isn't needed by `ninja all` or `ninja install`

Reply:
Correct. why should the test runner be built if the user won't execute the test suite?

Reply:
build_by_default is true by default though

Reply:
woops